doc/contrib/documentation.md

```diff
@@ -0,0 +1,137 @@
+
+In literature there is also goal-oriented but we prefer learning-oriented approaches.
+E.g. of course you might have the goal to write a new plugin.
+But why not also learn about plugins while creating a new plugin?
+
+> Takeaway:
+> Don't try to combine different orientations in one document, instead split your documentation up in e.g. a README.md (information), tutorial (learning), decisions (understanding).
+
+## Criteria
+
+Elektra's documentation must fulfill:
+
+- Always write what **is** not what you would like to have.
+  Explanations must always refer to the current situation (as changed with your PR).
+- It is self-contained.
+  It is not enough to link to some paper and an external page as explanation.
+  All explanation must be within the repository, e.g., in case the external information goes away.
+  This doesn't apply if the authoritative standard lives outside of Elektra.
+  Then you would write, e.g., "The toml plugin implements [this standard](https://toml.io/en/v1.0.0), with following extensions:".
+  The extensions, however, again must be fully listed within our repository.
+- We use standard Markdown where possible, with only a few extensions:
+  - styled fenced blocks
+  - `- [ ]` option lists
+  - `- <word>:<line break>` description lists
+- The documentation should be as near the code as possible.
+
+> Takeaway:
+> Include full API and Markdown documentation of the current situation directly in your PRs.
+
+## Style
+
+- Sentences are short and written in one line.
+  I.e. lines usually end with `.`, `:` or `;`.
+  Avoid line breaks in the middle of the sentence.
+- Use active and strong verbs early in your sentences.
+  "We" refers to the community around the Elektra Initiative.
+- Use headings and lists to keep a clear structure in the text.
+- Use examples and images to emphasize important points, don't overuse emphasis in text (bold, etc.).
+- Spelling is American English with [spellings as defined here](/scripts/sed).
+- It is consistent with our [terminology](/doc/help/elektra-glossary.md).
+
+> Note:
+> Please extend [terminology](/doc/help/elektra-glossary.md) and [spellings](/scripts/sed) as needed.
+
+## Completeness
+
+In general the documentation does not need to be complete.
+In particular, we do not want repetition of implementation details as documentation.
+[Prefer to write self-documenting code](/doc/CODING.md).
+Nevertheless, there are a few must-haves:
+
+- A README.md must be available for every module.
+- A man page (help/kdb-) must be available for every command (including external commands).
+- A tutorial must be present for every important concept.
+- Every documentation page must be listed in the [Website's structure](/src/tools/website/resources/structure.json.in).
+- Everything copied must be properly licensed in [reuse](/.reuse/dep5).
+
+## Links
+
+Generously use links but be very careful to create a red line through the documentation:
```

Comment:
I'm not 100% sure, but I think the phrase "red line/red thread" is a Germanism and probably not so well known outside of the German community?
```suggestion
Generously use links but be very careful to create a coherent documentation:
```